Formula/kyuubi.rb

```diff
@@ -0,0 +1,36 @@
+class Kyuubi < Formula
+  desc "Serverless SQL on data warehouses and lakehouses"
+  homepage "https://kyuubi.apache.org/"
+  version "1.6.1-incubating"
+  url "https://dlcdn.apache.org/kyuubi/kyuubi-#{version}/apache-kyuubi-#{version}-bin.tgz"
+  sha256 "a64a6e5d71b39b0e494597282493250df1c74714fc2bba5057946926c0a32ba2"
+  license "Apache-2.0"
+  head "https://github.com/apache/kyuubi.git", branch: "master"
+
+  depends_on "openjdk@8"
+
+  def create_bin_with_envs(target = "kyuubi", source = "kyuubi")
+    (bin/target).write_env_script libexec/"bin"/source,
+      JAVA_HOME:            Formula["openjdk@8"].opt_prefix.to_s,
+      SPARK_HOME:           Formula["apache-spark"].opt_libexec.to_s,
+      KYUUBI_HOME:          libexec.to_s,
+      KYUUBI_LOG_DIR:       var/"kyuubi/logs",
+      KYUUBI_PID_DIR:       var/"kyuubi/pid",
+      KYUUBI_WORK_DIR_ROOT: var/"kyuubi/work"
+  end
+
+  def install
+    libexec.install Dir["*"]
+    (var/"kyuubi/logs").mkpath
+    (var/"kyuubi/work").mkpath
+    (var/"kyuubi/pid").mkpath
+    create_bin_with_envs
+    # Rename the beeline to kyuubi-beeline to avoid conflicts with spark and hive
+    create_bin_with_envs "kyuubi-beeline", "beeline"
+  end
+
+  test do
+    assert_match "org.apache.kyuubi.server.KyuubiServer", shell_output("#{bin}/kyuubi stop")
```

Comment:
there are other examples in the repo that can help you with the server start/stop testing.

Reply:
Addressed other comments, will reference some examples to improve the testing tomorrow(UTC+8)